src/sdk/pynni/nni/compression/torch/utils/shape_dependency.py

```diff
@@ -185,6 +185,52 @@ def dependency_sets(self):
             d_sets.append(tmp_set)
         return d_sets
 
+class InputChannelDependency(ChannelDependency):
+    def __init__(self, model, dummy_input=None, traced_model=None):
+        super(InputChannelDependency, self).__init__(model, dummy_input, traced_model)
+
+    def _get_following_convs(self, tensor):
+        queue = []
+        key_layers = []
+        queue.extend(self.graph.input_to_node[tensor])
+        while queue:
+            curnode = queue.pop(0)
+            if curnode.op_type == 'Conv2d' or curnode.op_type == 'Linear':
```

Comment:
It seems linear is not added into the set.   I tested it, for resnet 18:  `layer4.1.conv1` and `fc` should be in the same set, but  `fc` is not added into the set.

Reply:
There is a flatten operation between the `fc` layer and the `layer4.1.conv1`, in the current version, the reshape version may cut the dependency.
![image](https://user-images.githubusercontent.com/49771382/93179208-ee4df500-f767-11ea-9bf7-b6311cd88b57.png)

Reply:
since we are pruning feature map on dim=1,  can we make flatten not break dependencies if second params is `1` ? `torch.flatten(x, 1)`

Reply:
Yes, you are right, we should check whether the parameters will break dependency~ I'll update, thanks~

Reply:
Updated~